components/Shipping/basic_usage.rst

```diff
@@ -0,0 +1,656 @@
+Basic Usage
+===========
+
+Shipping Category
+-----------------
+
+Every shipping category has two identifiers, an ID and name. You can access those by calling ``->getId()`` and ``->getName()``
+methods respectively. The name is mutable, so you can change it by calling ``->setName('Regular')`` on the shipping category instance.
+
+Shipping Method
+---------------
+
+Every shipping method has two identifiers, an ID and name. You can access those by calling ``->getId()`` and ``->getName()``
+methods respectively. The name is mutable, so you can change it by calling ``->setName('FedEx')`` on the shipping method instance.
+
+Setting Shipping Category
+~~~~~~~~~~~~~~~~~~~~~~~~~
+
+Every shipping method can have shipping category. You can simply set or unset it by calling ``->setCategory()``.
+
+.. code-block:: php
+
+    <?php
+
+        use Sylius\Component\Shipping\Model\ShippingMethod;
+        use Sylius\Component\Shipping\Model\ShippingCategory;
+        use Sylius\Component\Shipping\Model\ShippingMethodInterface;
+
+        $shippingCategory = new ShippingCategory();
+        $shippingCategory->setName('Regular'); // Regular weight items
+
+
+        $shippingMethod = new ShippingMethod();
+        $shippingMethod->setCategory($shippingCategory); //default null, detach
+        $shippingMethod->getCategory(); // Output will be ShippingCategory object
+        $shippingMethod->setCategory(null);
+
+Setting Rule
+~~~~~~~~~~~~
+
+Every shipping method can have many rules, which defines additional requirements for shipment. If shipment do not
+fulfil this requirements (for example rule sets expected quantity of shipment items to 2, but shipment has only one **ShipmentItem**)
+then shippingMethod with this rule cannot be used to this shipment.
+
+.. code-block:: php
+
+    <?php
+
+        use Sylius\Component\Shipping\Model\Rule;
+        use Sylius\Component\Shipping\Model\ShippingMethod;
+
+        $shippingMethod = new ShippingMethod();
+        $rule = new Rule();
+
+        $shippingMethod->addRule($rule);
+        $shippingMethod->hasRule($rule); // returns true
+        $shippingMethod->getRules(); // collection of rules with count equals 1
+        $shippingMethod->removeRule($rule);
+        $shippingMethod->hasRule($rule); // returns false
+
+Shipping Method Translation
+---------------------------
+
+**ShippingMethodTranslation** allows translate name of shipping method according to the given locales. To see how to use translation
+please go to :ref:`component_translation_basic-translations`.
+
+Rule
+----
+
+**Rule** defines additional requirements for shipment method, which should be fulfil by shipment if it have to be delivered in
```

Comment:
`A **Rule** defines additional requirements for a **ShippingMethod**, which have to be fulfilled by the **Shipment**, if it has to be delivered in a way described by this **ShippingMethod**.`